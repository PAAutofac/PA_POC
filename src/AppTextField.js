import React ,{Fragment} from 'react'
import { makeStyles } from '@material-ui/core/styles';
import {TextField} from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
    textfieldControl: {
      margin: theme.spacing(3), 
    },
  }));

export const AppTextField = (props) =>{
    const classes = useStyles();
    return (
    <Fragment>
    <TextField id={props.id} label={props.label} type={props.type} style={{ width: props.width }}  className={classes.textfieldControl} />
    </Fragment>
    )
}