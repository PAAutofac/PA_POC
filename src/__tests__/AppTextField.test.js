// __tests__/CheckboxWithLabel-test.js
import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import {AppTextField} from '../AppTextField'
import renderer from 'react-test-renderer';
import { act } from "react-dom/test-utils";

let container = null;

beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

// Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
// unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

it('Textfield test 1', () => {
  const component = renderer.create(
    <AppTextField id="Test1" label="Test text field" type="number" width="10%" />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders with or without a value", () => {
    act(() => {
      render(<AppTextField />, container);
    });
    expect(container.textContent).toBe("");
    
  });
