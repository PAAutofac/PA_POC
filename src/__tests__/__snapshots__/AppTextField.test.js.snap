// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield test 1 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-textfieldControl-1"
  style={
    Object {
      "width": "10%",
    }
  }
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
    htmlFor="Test1"
    id="Test1-label"
  >
    Test text field
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      id="Test1"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="number"
    />
  </div>
</div>
`;
